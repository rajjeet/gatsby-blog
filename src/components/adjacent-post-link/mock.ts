export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
