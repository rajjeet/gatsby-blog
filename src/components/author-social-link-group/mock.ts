import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

