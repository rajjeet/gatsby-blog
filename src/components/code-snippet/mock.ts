import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
