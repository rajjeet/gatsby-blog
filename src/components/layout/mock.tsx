import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

