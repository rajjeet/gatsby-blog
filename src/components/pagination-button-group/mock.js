export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
