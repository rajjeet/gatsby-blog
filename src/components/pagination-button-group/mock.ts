export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}) => ({
  className: {}, currentPage, numOfPages, paginationSlug: '',
});
