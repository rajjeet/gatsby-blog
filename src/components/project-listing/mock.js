import { createMockProjects } from '../../utils/testing';

export const makeProps = () => ({
  className: {}, projects: createMockProjects, heading: '',
});
