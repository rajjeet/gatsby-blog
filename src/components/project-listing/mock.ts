import { createMockProjects } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
  projects: createMockProjects,
  heading: 'List of Projects',
});

