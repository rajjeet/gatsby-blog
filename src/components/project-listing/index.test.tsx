import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ProjectListing } from './index';
import { makeProps } from './mock';

afterEach(cleanup);

describe('<ProjectListing />', () => {
  it('should render', () => {
    const { asFragment } = render(<ProjectListing {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a heading', () => {
    const { getByText } = render(<ProjectListing {...makeProps()} />);
    expect(getByText('List of Projects')).toBeDefined();
  });

  it('should display the list of projects', () => {
    const { getByText } = render(<ProjectListing {...makeProps()} />);
    expect(getByText('Condo Landing Pages')).toBeDefined();
    expect(getByText('Kitchen Quoter')).toBeDefined();
  });
});
