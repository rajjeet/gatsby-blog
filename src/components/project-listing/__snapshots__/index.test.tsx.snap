// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectListing /> should render 1`] = `
<DocumentFragment>
  <div
    class="[object Object] sc-bxivhb nYNVZ"
  >
    <h1>
      List of Projects
    </h1>
    <div
      class="sc-htpNat cDaIEp"
    >
      <a
        aria-label="See Condo Landing Pages"
        class="sc-bwzfXH kTyWdp"
        href="/projects/condo-landing-pages/"
      >
        <h3>
          Condo Landing Pages
        </h3>
        <div
          class="sc-bdVaJa cBDXGb gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <picture>
            <source
              sizes=""
              srcset=""
            />
            <img
              alt="Condo Landing Pages"
              loading="lazy"
              sizes=""
              src=""
              srcset=""
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="" alt="Condo Landing Pages" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <p>
          An automated process for building fast, responsive landing pages for pre-construction condos.
        </p>
      </a>
    </div>
    <div
      class="sc-htpNat cDaIEp"
    >
      <a
        aria-label="See Kitchen Quoter"
        class="sc-bwzfXH kTyWdp"
        href="/projects/kitchen-quoter/"
      >
        <h3>
          Kitchen Quoter
        </h3>
        <div
          class="sc-bdVaJa cBDXGb gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <picture>
            <source
              sizes=""
              srcset=""
            />
            <img
              alt="Kitchen Quoter"
              loading="lazy"
              sizes=""
              src=""
              srcset=""
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="" alt="Kitchen Quoter" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <p>
          A do-it-yourself calculator that generates dollar quotes based on user-specified fields and formula.
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
